From 6a8dbe4b7979d058d2e58b89953392d8ff85f973 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:29:37 -0400
Subject: [net] openvswitch: rename ->sync to ->syncp

Message-id: <0032e8b7f6fc9995bebf7459cbc1d3881326560f.1410531783.git.jbenc@redhat.com>
Patchwork-id: 93992
O-Subject: [RHEL7.1 net PATCH 04/50] openvswitch: rename ->sync to ->syncp
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

RHEL backport notes: not strictly needed, as we don't have nor need
netdev_alloc_pcpu_stats but lowers difference from upstream and does not
cause any harm.

commit df9d9fdf8fdad710949ce52a403684c991ced29b
Author: WANG Cong <xiyou.wangcong@gmail.com>
Date:   Fri Feb 14 15:10:46 2014 -0800

    openvswitch: rename ->sync to ->syncp

    Openvswitch defines u64_stats_sync as ->sync rather than ->syncp,
    so fails to compile with netdev_alloc_pcpu_stats(). So just rename it to ->syncp.

    Reported-by: kbuild test robot <fengguang.wu@intel.com>
    Fixes: 1c213bd24ad04f4430031 (net: introduce netdev_alloc_pcpu_stats() for drivers)
    Cc: David S. Miller <davem@davemloft.net>
    Signed-off-by: Cong Wang <xiyou.wangcong@gmail.com>
    Reviewed-by: Flavio Leitner <fbl@redhat.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/openvswitch/datapath.c b/net/openvswitch/datapath.c
index a00e19f..862a42b 100644
--- a/net/openvswitch/datapath.c
+++ b/net/openvswitch/datapath.c
@@ -256,10 +256,10 @@ void ovs_dp_process_received_packet(struct vport *p, struct sk_buff *skb)
 
 out:
  /* Update datapath statistics. */
- u64_stats_update_begin(&stats->sync);
+ u64_stats_update_begin(&stats->syncp);
  (*stats_counter)++;
  stats->n_mask_hit += n_mask_hit;
- u64_stats_update_end(&stats->sync);
+ u64_stats_update_end(&stats->syncp);
 }
 
 static struct genl_family dp_packet_genl_family = {
@@ -295,9 +295,9 @@ int ovs_dp_upcall(struct datapath *dp, struct sk_buff *skb,
 err:
  stats = this_cpu_ptr(dp->stats_percpu);
 
- u64_stats_update_begin(&stats->sync);
+ u64_stats_update_begin(&stats->syncp);
  stats->n_lost++;
- u64_stats_update_end(&stats->sync);
+ u64_stats_update_end(&stats->syncp);
 
  return err;
 }
@@ -612,9 +612,9 @@ static void get_dp_stats(struct datapath *dp, struct ovs_dp_stats *stats,
   percpu_stats = per_cpu_ptr(dp->stats_percpu, i);
 
   do {
-   start = u64_stats_fetch_begin_bh(&percpu_stats->sync);
+   start = u64_stats_fetch_begin_bh(&percpu_stats->syncp);
    local_stats = *percpu_stats;
-  } while (u64_stats_fetch_retry_bh(&percpu_stats->sync, start));
+  } while (u64_stats_fetch_retry_bh(&percpu_stats->syncp, start));
 
   stats->n_hit += local_stats.n_hit;
   stats->n_missed += local_stats.n_missed;
diff --git a/net/openvswitch/datapath.h b/net/openvswitch/datapath.h
index 6be9fbb..0531738 100644
--- a/net/openvswitch/datapath.h
+++ b/net/openvswitch/datapath.h
@@ -55,7 +55,7 @@ struct dp_stats_percpu {
  u64 n_missed;
  u64 n_lost;
  u64 n_mask_hit;
- struct u64_stats_sync sync;
+ struct u64_stats_sync syncp;
 };
 
 /**
-- 
1.7.1